sdk/appcenter/src/main/java/com/microsoft/appcenter/utils/context/AuthTokenContext.java

```diff
@@ -352,16 +357,25 @@ public synchronized void removeOldestTokenIfMatching(String token) {
      * @param authTokenInfo auth token to check for expiration.
      */
     public void checkIfTokenNeedsToBeRefreshed(@NonNull AuthTokenInfo authTokenInfo) {
+        String accountId = getAccountIdToRefreshAndUpdateCache(authTokenInfo);
+        if (accountId != null) {
+            RefreshListener refreshListener = mRefreshListener.get();
+            if (refreshListener != null) {
```

Comment:
Please check this at the first place to make sure that any listener will handle already saved `mLastTokenRefreshed` value